service/tablet_allocator.hh

```diff
@@ -22,7 +22,22 @@ struct tablet_migration_info {
     locator::tablet_replica dst;
 };
 
-using migration_plan = utils::chunked_vector<tablet_migration_info>;
+class migration_plan {
+    utils::chunked_vector<tablet_migration_info> _migrations;
+public:
+    bool empty() const { return _migrations.empty(); }
+    auto begin() const { return _migrations.begin(); }
+    auto end() const { return _migrations.end(); }
+    size_t size() const { return _migrations.size(); }
```

Comment:
You can inherit from  `utils::chunked_vector<tablet_migration_info>` to get those automatically.

Reply:
In this case I think composition is more appropriate in the long term, because there is no longer an "is-a" relationship. It can get worse as we add more things to the plan, in which case chunked_vector methods will stop making sense, e.g. `external_memory_usage()` or `clear()`. 

Maybe I should add a getter for migrations() instead of this.

Reply:
Agree better to add a getter for migrations than to pretend mirgation_plan is a container.